src/Microsoft.ML.ImageAnalytics/ImageType.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using Microsoft.ML.Runtime.Data;
+
+namespace Microsoft.ML.Runtime.ImageAnalytics
+{
+    public sealed class ImageType: StructuredType
+    {
+        public readonly int Height;
+        public readonly int Width;
+        public ImageType(int height, int width)
+           : base(typeof(Bitmap))
+        {
+            Contracts.CheckParam(height > 0, nameof(height));
+            Contracts.CheckParam(width > 0, nameof(width));
+            Contracts.CheckParam((long)height * width <= int.MaxValue / 4, nameof(height), "height * width is too large");
+            Height = height;
+            Width = width;
+        }
+
+        public ImageType()
+            : base(typeof(Image))
+        {
+        }
+
+        public override bool Equals(ColumnType other)
```

Comment:
>Equals [](start = 29, length = 6)

There's something a bit strange about this. `ColumnType` implements `IEquatable<ColumnType>`, and the [guidelines for that](https://docs.microsoft.com/en-us/dotnet/api/system.iequatable-1?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev15.query%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(System.IEquatable%601);k(SolutionItemsProject);k(DevLang-csharp)%26rd%3Dtrue%26f%3D255%26MSPPError%3D-2147217396&view=netframework-4.7.2) say it should also overload `Equals(object)` and `GetHashCode`.

Reply:
Maybe file a separate issue, we can address later?

---
In reply to: [202379905](https://github.com/dotnet/machinelearning/pull/528#discussion_r202379905) [](ancestors = 202379905)

Reply:
#547

---
In reply to: [203182127](https://github.com/dotnet/machinelearning/pull/528#discussion_r203182127) [](ancestors = 203182127,202379905)